core/src/pretty.rs

```diff
@@ -755,10 +755,7 @@ where
                 .append(allocator.hardline()),
             // TODO
             Sealed(_i, _rt, _lbl) => allocator.text("#<sealed>").append(allocator.hardline()),
-            Annotated(annot, rt) => allocator
-                .atom(rt)
-                .append(allocator.space())
-                .append(allocator.annot(annot)),
+            Annotated(annot, rt) => allocator.atom(rt).append(allocator.annot(annot)),
```

Comment:
Wasn't the space necessary here? Or did it move to `allocator.annot` ?

Reply:
`annot_part` adds a `line()` before a type or contract annotation starts. That produces a space if everything in the current group fits on a line and a linebreak otherwise. So the space here was actually producing a double space before a contract assertion. On master:
```
❯ nickel pprint-ast <<<'foo | Number'
foo  | Number
```